### PR TITLE
Remove CMSIS-DAP which is confused with DAPLink

### PR DESCRIPTION
**Pull request by tz-arm**
Remove CMSIS-DAP which is confused with DAPLink, otherwise we need give more description for CMSIS-DAP and the difference between CMSIS-DAP and DAPLink.

### CODE REVIEW

File: docs/advanced/DAP.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# CMSIS-DAP and DAPLink
+# DAPLink
 
 
 ## DAPLink
```
Comment (AnotherButler) [inline docs/advanced/DAP.md]: Please delete the second heading, so we don't have two headings in a row that say the same thing

**Comment by AnotherButler** [comment]
@tz-arm Thank you for the PR. This is a good point. I left one comment.

File: docs/advanced/DAP.md (hunk 1 above)
Comment (tz-arm) [inline docs/advanced/DAP.md]: Done, thanks for your review and comment!

Comment (AnotherButler) [inline docs/advanced/DAP.md]: Thanks 👍

**Pull request merged**